### PR TITLE
Added weapon tints

### PR DESCRIPTION
**Pull request by Profex1999**
2 new methods for xPlayer
-getWeaponTint(weaponName)
-setWeaponTint(weaponName)
The tint is saved in DB 
If pickup type is a weapon, tint is also applied on the object

### CODE REVIEW

File: server/classes/player.lua [hunk 1]
```diff
@@ -375,14 +375,31 @@ function CreateExtendedPlayer(player, accounts, inventory, job, loadout, name, c
...
+		local loadoutNum, weapon = self.getWeapon(weaponName)
+
+		if weapon then
+			return weapon.tint
+		end
+	end
+
+	self.setWeaponTint = function(weaponName, index)
+		local loadoutNum, weapon = self.getWeapon(weaponName)
+		if weapon then
```
Comment (ElPumpo) [inline server/classes/player.lua]: missing whitespace

File: client/main.lua [hunk 2]
```diff
@@ -183,6 +183,11 @@ AddEventHandler('esx:addWeaponComponent', function(weaponName, weaponComponent)
 	GiveWeaponComponentToPed(playerPed, weaponHash, componentHash)
 end)
 
+RegisterNetEvent('esx:setWeaponTint')
+AddEventHandler('esx:setWeaponTint', function(weaponName, index)
+	SetPedWeaponTintIndex(PlayerPedId(), GetHashKey(weaponName), index)
```
Comment (ElPumpo) [inline client/main.lua]: maybe rename the `index` argument to `tintIndex`

File: client/main.lua [hunk 3]
```diff
@@ -86,7 +86,7 @@ AddEventHandler('esx:restoreLoadout', function()
 
 		GiveWeaponToPed(playerPed, weaponHash, 0, false, false)
 		local ammoType = GetPedAmmoTypeFromWeapon(playerPed, weaponHash)
-
+		SetPedWeaponTintIndex(playerPed, weaponHash, v.tint)
```
Comment (ElPumpo) [inline client/main.lua]: Why is the whitespace removed, add it after the new line added

File: server/functions.lua [hunk 4]
```diff
@@ -159,7 +159,7 @@ ESX.GetItemLabel = function(item)
 	end
 end
 
-ESX.CreatePickup = function(type, name, count, label, playerId, components)
+ESX.CreatePickup = function(type, name, count, label, playerId, components, tint)
```
Comment (ElPumpo) [inline server/functions.lua]: rename this argument to `tintIndex` too

**Comment by ElPumpo** [comment]
Also on xplayer creation you'll have to add compability for old saves in db.

**Comment by Profex1999** [comment]
> Also on xplayer creation you'll have to add compability for old saves in db.
Do you mean like this?

**Comment by ElPumpo** [comment]
Yes.

**Comment by ElPumpo** [comment]
But you missed a whitespace in your latest commit.. again

**Comment by ElPumpo** [comment]
But we will now consider your pull request. We can also implement it like we did components, which has an configuration file that contains a label, hash and name. We can do something similar with tints (label, tintIndex, name) but I'll have to do some research about it.

I dont like the idea of storing the raw tint index..

**Comment by Profex1999** [comment]
Yeah, I used the raw index because the range (0-8) is the same for all weapons (instead of components that have a different hash for every weapon).  Maybe we could put the colors generally in the config (since they are the same for every weapon). I was also working on adding MK 2 weapons compatibility and so things would go different because MK2 weapons has like 33 colors instead of 8.

**Comment by ElPumpo** [comment]
Yes after some quick research it seems like all MK2 weapons have an extended tint range, ranging from 0 (default) to 31

**Comment by ElPumpo** [comment]
Not every weapon supports weapon tints, I'm working on a similar implementation using an configuration file for supported tints.

Seems like most weapons that support tints have the same colors for each tint index, but there are weapons where this doesnt apply and because of that the best implementation would be having a list for each tint, almost like components. Either have a key-value table or index-value..

**Pull request closed without merging**